src/app/pages/communication/communication.page.ts

```diff
@@ -1,30 +1,354 @@
+  getCandidateDriverNumber,
+  getCandidateEmailAddress,
+  getCandidateName,
+  getPostalAddress,
+  getUntitledCandidateName,
+} from '@store/tests/journal-data/common/candidate/candidate.selector';
+import { map, take, tap } from 'rxjs/operators';
+import { getCommunicationPreference } from '@store/tests/communication-preferences/communication-preferences.reducer';
+import {
+  getCommunicationPreferenceType,
+  getCommunicationPreferenceUpdatedEmail,
+  getConductedLanguage,
+} from '@store/tests/communication-preferences/communication-preferences.selector';
+import { Language } from '@store/tests/communication-preferences/communication-preferences.model';
+import { configureI18N } from '@shared/helpers/translation.helpers';
+import * as communicationPreferencesActions
+  from '@store/tests/communication-preferences/communication-preferences.actions';
+import {
+  CommunicationSubmitInfo,
+  CommunicationSubmitInfoError,
+  CommunicationValidationError,
+} from '@pages/communication/communication.actions';
+import { DeviceAuthenticationProvider } from '@providers/device-authentication/device-authentication';
+import { getTestCategory } from '@store/tests/category/category.reducer';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
 import { TestFlowPageNames } from '@pages/page-names.constants';
 
+interface CommunicationPageState {
+  candidateName$: Observable<string>;
+  candidateUntitledName$: Observable<string>;
+  candidateDriverNumber$: Observable<string>;
+  candidateProvidedEmail$: Observable<string>;
+  communicationEmail$: Observable<string>;
+  communicationType$: Observable<string>;
+  candidateAddress$: Observable<Address>;
+  conductedLanguage$: Observable<string>;
+  testCategory$: Observable<CategoryCode>;
+}
+
 @Component({
   selector: 'app-communication',
   templateUrl: './communication.page.html',
   styleUrls: ['./communication.page.scss'],
 })
-export class CommunicationPage implements OnInit {
+export class CommunicationPage extends PracticeableBasePageComponent implements OnInit {
+
+  static readonly providedEmail: string = 'Provided';
+  static readonly updatedEmail: string = 'Updated';
+  static readonly email: CommunicationMethod = 'Email';
+  static readonly post: CommunicationMethod = 'Post';
+  static readonly notProvided: CommunicationMethod = 'Not provided';
+  static readonly welshLanguage: ConductedLanguage = 'Cymraeg';
+  static readonly englishLanguage: ConductedLanguage = 'English';
+
+  form: FormGroup;
+  subscription: Subscription;
+  emailType: string;
+  pageState: CommunicationPageState;
+  candidateProvidedEmail: string;
+  communicationEmail: string;
+  communicationType: CommunicationMethod;
+  merged$: Observable<string | boolean>;
+  testCategory: CategoryCode;
 
   constructor(
-    private navController: NavController,
+    public platform: Platform,
+    public router: Router,
+    public authenticationProvider: AuthenticationProvider,
```

Comment:
platform, router and authenticationProvider can have their access modifiers removed if just being used to inject into basepage

Reply:
@andrewsetterfield The same comment was added to MES-6277